src/s3/commands/uploadFile.ts

```diff
@@ -114,11 +148,221 @@ async function uploadWithProgress({
+        const response = verifySinglePickerOutput(await promptUser({
+            picker: picker,
+            onDidTriggerButton: (button, resolve, reject) => {
+                if (button === vscode.QuickInputButtons.Back) {
+                    resolve(undefined)
+                }
+            },
+        }))
+
+        if(!response){
+            getLogger().info('UploadFile cancelled')
+            telemetry.recordS3UploadObject({ result: 'Cancelled' })
+            return
+        }
+
+        if (response!.label === 'Browse for more files...') {
+            selectAnotherFile = true
+        }
+
+    //2.1 step: if user wants other file prompt the file selector to allow user to select wanted file
+        if (selectAnotherFile) {
+            fileLocation = await promptForFileLocation(Window.vscode())
+        }
+    }
+    //if at this point the file location is undefined, something went wrong, cancel the upload
+    if (!fileLocation) {
+        getLogger().info('UploadFile cancelled')
+        telemetry.recordS3UploadObject({ result: 'Cancelled' })
+        return
+    }
+
+    
+    //3 step: get and display the user's available buckets --> goal is to get node: S3BucketNode | S3FolderNode
+    //const node: S3BucketNode | S3FolderNode = promptUserForBucket()
+    const bucket: S3.Bucket | undefined = await promptUserForBucket()
+    if(!bucket){
+        getLogger().info('No bucket selected, cancelling upload')
+        telemetry.recordS3UploadObject({ result: 'Cancelled' })
+        return
+    }
+    
+    const fileName = path.basename(fileLocation.fsPath)
+    const key = '' + fileName
+    const destinationPath = readablePath({ bucket: {name: bucket.Name!}, path: key })
+    const regionCode = Workspace.vscode().getConfiguration(extensionSettingsPrefix).get<string>('s3.defaultRegion')?? "us-east-1"
+    //4 step: after user selects bucket, upload file to that bucket and show progress
+    try {
+        showOutputMessage(`Uploading file from ${fileLocation} to ${destinationPath}`, outputChannel)
+        const request = {
+            bucketName: bucket.Name!,
+            key: key,
+            fileLocation: fileLocation,
+            fileSizeBytes: fileSizeBytes(fileLocation),
+            s3Client: ext.toolkitClientBuilder.createS3Client(regionCode),
+            window: window
+        }
+
+        await uploadWithProgress(request)
+        showOutputMessage(`Successfully uploaded file ${destinationPath} to ${bucket.Name!}`, outputChannel)
+        telemetry.recordS3UploadObject({ result: 'Succeeded' })
+
+    } catch (e) {
+        getLogger().error(`Failed to upload file from ${fileLocation} to ${destinationPath}: %O`, e)
+        showErrorWithLogs(localize('AWS.s3.uploadFile.error.general', 'Failed to upload file {0}', fileName), window)
+        telemetry.recordS3UploadObject({ result: 'Failed' })
+    }
+    //await refreshNode(bucket, commands)
```

Comment:
Assuming this is commented because you won't be able to infer a node? Consider refreshing the whole AWS explorer instead.